client/src/context/useSchedulerContext.tsx

```diff
@@ -0,0 +1,115 @@
+import { useState, useContext, createContext, FunctionComponent, useEffect, useCallback } from 'react';
+
+import moment from 'moment-timezone';
+
+// schedule is meant to store availability from the google calendar api
+// availableDays is meant to store the days and hours available as set by the user from the onboard page
+
+interface ISchedulerContext {
+  selectedDay: Date | undefined;
+  updateSelectedDayContext: (day: Date | undefined) => void;
+  timezone: string | undefined;
+  updateTimezoneContext: (tz: string) => void;
+  schedule: Array<Date> | undefined;
+  updateScheduleContext: (calendar: Array<Date> | undefined) => void;
+  availability: Record<string, Array<string>>;
+  updateAvailabilityContext: (hours: Record<string, Array<string>>) => void;
+  modal: boolean;
+  updateModalContext: (open: boolean) => void;
+  meeting: string;
+  updateMeetingContext: (time: string) => void;
+}
+
+export const SchedulerContext = createContext<ISchedulerContext>({
+  selectedDay: undefined,
+  updateSelectedDayContext: () => null,
+  timezone: undefined,
+  updateTimezoneContext: () => null,
+  schedule: [],
+  updateScheduleContext: () => null,
+  availability: {
+    Sunday: [],
+    Saturday: [],
+    Monday: [],
+    Tuesday: [],
+    Wednesday: [],
+    Thursday: [],
+    Friday: [],
+  },
+  updateAvailabilityContext: () => null,
+  modal: false,
+  updateModalContext: () => null,
+  meeting: '',
+  updateMeetingContext: () => null,
+});
+
+export const SchedulerProvider: FunctionComponent = ({ children }): JSX.Element => {
+  const [selectedDay, setSelectedDay] = useState<Date | undefined>(undefined);
+  const [timezone, setTimezone] = useState<string | undefined>('');
+  const [schedule, setSchedule] = useState<Array<Date> | undefined>([]);
+  const [availability, setAvailability] = useState<Record<string, Array<string>>>({
+    Sunday: [],
+    Saturday: [],
+    Monday: ['placeholder', 'placeholder'],
+    Tuesday: ['placeholder', 'placeholder'],
+    Wednesday: ['placeholder', 'placeholder'],
+    Thursday: ['placeholder', 'placeholder'],
+    Friday: ['placeholder', 'placeholder'],
+  });
+  const [modal, setModal] = useState<boolean>(false);
+  const [meeting, setMeeting] = useState<string>('');
+
+  const updateSelectedDayContext = useCallback((day: Date | undefined) => {
+    setSelectedDay(day);
+  }, []);
```

Comment:
Could you elaborate on the purpose for using `useCallback` here? Thank you!

Reply:
That's a mistake on my part honesty. I just looked at the other context files and copy pasted them while changing up what I thought I was necessary. I'll clean that up

Reply:
The only part that actually makes sense coincidentally is the callback hook for the updateTimezoneContext where we don't want the function running every single time on re-render because of the useEffect. We only want to run it once on initial render unless the user specifically changes the timezone themselves.